pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -329,6 +386,12 @@ func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Contai
 
 		availablePhysicalCPUs := p.GetAvailablePhysicalCPUs(s).Size()
 
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			if cs, ok := podutil.GetContainerStatus(pod.Status.ContainerStatuses, container.Name); ok {
+				cpuAllocatedQuantity := cs.AllocatedResources[v1.ResourceCPU]
```

Comment:
are we guaranteed to have CPU resources here? Or, equivalently, can we handle the zero value correctly?

Reply:
Hi now with the e2e_node tests in place, I feel more confident will dig more into this.

Reply:
Hi I think that since we depend solely on topology manager and we haven't touched this part of the code, CPU resources are guaranteed from this point of view, otherwise it should fail with error. For the API considering the latest updates discussed in InPlacePodVerticalScaling we will need to investigate this again with the new API,  this part will need to be reevaluated and updated.